core/src/main/java/tc/oc/pgm/score/ScoreModule.java

```diff
@@ -94,25 +94,24 @@ public ScoreModule parse(MapFactory factory, Logger logger, Document doc)
       }
 
       RegionParser regionParser = factory.getRegions();
-      ScoreConfig config = new ScoreConfig();
       ImmutableSet.Builder<ScoreBoxFactory> scoreBoxFactories = ImmutableSet.builder();
-
+      ScoreConfig config = null;
       for (Element scoreEl : scoreElements) {
-        config.scoreLimit = XMLUtils.parseNumber(scoreEl.getChild("limit"), Integer.class, -1);
-        config.mercyLimit = XMLUtils.parseNumber(scoreEl.getChild("mercy"), Integer.class, -1);
+        int scoreLimit = XMLUtils.parseNumber(scoreEl.getChild("limit"), Integer.class, -1);
+        int mercyLimit = XMLUtils.parseNumber(scoreEl.getChild("mercy"), Integer.class, -1);
 
         // For backwards compatibility, default kill/death points to 1 if proto is old and <king/>
         // tag
         // is not present
         boolean scoreKillsByDefault =
             proto.isOlderThan(MapProtos.DEFAULT_SCORES_TO_ZERO) && scoreEl.getChild("king") == null;
-        config.deathScore =
+        int deathScore =
             XMLUtils.parseNumber(
                 scoreEl.getChild("deaths"), Integer.class, scoreKillsByDefault ? 1 : 0);
-        config.killScore =
+        int killScore =
             XMLUtils.parseNumber(
                 scoreEl.getChild("kills"), Integer.class, scoreKillsByDefault ? 1 : 0);
-
+        config = new ScoreConfig(scoreLimit, deathScore, killScore, mercyLimit);
```

Comment:
This is a good change (making ScoreConfig immutable), however this implementation still has issues

```xml
<score>
    <limit>100</limit>
</score>
<score>
    <mercy>3</mercy>
</score>
```

This is completely valid syntax, one iteration of the loop sets the limit to 100, the next iteration should set mercy to 3, and the map should get both properties, however, both current implementation and the modified one carry the same issue, they keep the values of only the last score element (resetting the values of everything else).

Later PGM had fixed this here <https://github.com/OvercastNetwork/ProjectAres/blob/master/PGM/src/main/java/tc/oc/pgm/score/ScoreModule.java#L98>

While porting the newer implementation is probably a bit overkill, it would be nice to move the 4 ints to outside the loop and update them inside, as well as passing the previous value as default for parseNumber, that way the values incrementally get updated. Then at the end create the ScoreConfig outside the loop.